Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -49,94 +68,299 @@ def Get_Marshall(Ensure):
+                desired_plugin_state = CurrentPluginState
+        # If there are any other workspaces with auoms outconf files, then
+        # we don't want to remove the audit rules
+        if other_ws_conf_count > 0:
+            desired_rules = GetDesiredAuditRules(AuditVersion)
+
+    return (desired_plugin_state, desired_rules, desired_outconf, desired_plugin_conf)
+
+def GetEnsure(WorkspaceId):
+    (audit_version, plugin_state,
+     actual_audit_rules, actual_outconf, actual_plugin_conf) = GetState(WorkspaceId)
+    if audit_version is None:
+        return None
+
+    Ensure = "Absent"
+    if (not actual_audit_rules is None and
+            not actual_outconf is None and
+            not actual_plugin_conf is None):
+        Ensure = "Present"
 
-def IsPluginEnabled():
+    (desired_plugin_state, desired_rules, desired_outconf,
+     desired_plugin_conf) = GetDesiredState(WorkspaceId, Ensure, audit_version, plugin_state)
+    if desired_plugin_state is None:
+        return None
+
+    Ensure = "Present"
+
+    if desired_plugin_state != plugin_state:
+        Ensure = "Absent"
+        LG().Log(LOG_INFO, "Plugin State does not match desired state")
+
+    if IsTextDifferent(desired_rules, actual_audit_rules):
+        Ensure = "Absent"
+        LG().Log(LOG_INFO, "Audit Rules do not match desired rules")
+
+    if IsTextDifferent(desired_outconf, actual_outconf):
+        Ensure = "Absent"
+        LG().Log(LOG_INFO, "auoms outconf do not match desired conf")
+
+    if IsTextDifferent(desired_plugin_conf, actual_plugin_conf):
+        Ensure = "Absent"
+        LG().Log(LOG_INFO, "Plugin Conf do not match desired conf")
+
+    return Ensure
+
+def GetState(WorkspaceId):
     args = []
     args.append("/usr/bin/sudo")
     args.append("/opt/microsoft/omsconfig/Scripts/OMSAuditdPlugin.sh")
-    args.append("test")
+    args.append("get")
+    args.append(GetTmpDir(WorkspaceId))
     proc = subprocess.Popen(args, -1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     exit_code = proc.wait()
-    if exit_code < 0 or exit_code > 2:
-        out_txt = proc.stdout.read().decode('utf8','replace').encode('utf8','replace')
-        err_txt = proc.stderr.read().decode('utf8','replace').encode('utf8','replace')
-        LG().Log('INFO', "stdout: " + out_txt)
-        LG().Log('INFO', "stderr: " + err_txt)
-        return None
-
+    out_txt = proc.stdout.read().decode('utf8', 'replace').encode('utf8', 'replace')
+    if exit_code != 0:
+        err_txt = proc.stderr.read().decode('utf8', 'replace').encode('utf8', 'replace')
+        LG().Log(LOG_ERROR, "Command failed: " + " ".join(args))
+        LG().Log(LOG_INFO, "stdout: " + out_txt)
+        LG().Log(LOG_INFO, "stderr: " + err_txt)
+        if exit_code == 2:
+            LG().Log(LOG_FATAL, "Unknown error while trying to determine OMSAuditdPlugin state")
+        else:
+            LG().Log(LOG_FATAL, "Could not determine auditd version: Perhaps it isn't installed")
```

Comment:
Seems to me that these two statements should be switched. A known exit code should not correspond to an unknown error, while the else cases prints out a specific message.